src/components/MentorList/useWarnings.tsx

```diff
@@ -0,0 +1,37 @@
+import { useState, useEffect } from 'react';
+import { getWarningsById } from '../../lib/calomentor';
+
+export function useWarnings(id: string) {
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const [isError, setIsError] = useState<boolean>(false);
+  const [warnings, setWarnings] = useState<number>(0);
+  const [mentorships, setMentorships] = useState<number>(0);
+
+  const getWarnings = async () => {
+    let response: Response;
+    try {
+      response = await getWarningsById(id);
+    } catch (error) {
+      setIsError(true);
+      setIsLoading(false);
+    }
+
+    if (!response?.ok) {
+      setIsError(true);
+      setIsLoading(false);
+    }
+
+    const { data } = await response.json();
+
+    setWarnings(data?.active_warnings);
+    setMentorships(data?.mentorships);
+    setIsLoading(false);
+  };
+
+  useEffect(() => {
+    getWarnings();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
This is a no-go, the lint rule avoid logic errors.

Check "my function value is constant" [in this comment](https://github.com/facebook/react/issues/14920#issuecomment-471070149).

Function values are not constants, because they close over values of the context around them (if some or all of the the values of which the function closes over change, the function will change also).

The simplest fix is to create the function **inside** the effect, which also has the benefit would be more performant because the function would be created only on response to id changes (although that is not really important, the difference would be minimal).

Reply:
or can wrap the getWarnings in `useCallback` if you need and pass the getWarnings as a dependency and remove the `id`